### PR TITLE
Fixed sorting issue with concurrent dictionary

### PR DESCRIPTION
**Pull request by fenix2222**
As per issue #1420

### CODE REVIEW

**Comment by dangershony** [comment]
I don't think it matters so much the time of the snapshot, we need to prevent concurrency bugs.
But ok use `ToArray()` or if you ant to use to `ToList()` then enumerate the first item as so
```
this.peers.Skip(0).ToList()
```

File: src/Stratis.Bitcoin/P2P/PeerAddressManager.cs [hunk 1]
```diff
@@ -158,7 +158,13 @@ public void SavePeers()
                 return;
 
             var fileStorage = new FileStorage<List<PeerAddress>>(this.PeerFilePath.AddressManagerFilePath);
-            fileStorage.SaveToFile(this.peers.OrderByDescending(p => p.Value.LastConnectionSuccess).Select(p => p.Value).ToList(), PeerFileName);
+            fileStorage.SaveToFile(
+                this.peers
+                    .ToList()
+                    .OrderByDescending(p => p.Value.LastConnectionSuccess)
+                    .Select(p => p.Value)
+                    .ToList(),
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerAddressManager.cs]: You probably don't need to enumerate twice so remove the second `ToList()`

Comment (dangershony) [reply src/Stratis.Bitcoin/P2P/PeerAddressManager.cs]: Perhaps also for readability take the snapshot first in a new param?

Comment (fenix2222) [reply src/Stratis.Bitcoin/P2P/PeerAddressManager.cs]: Actually looking at the implementation, `ToArray` is definitely safer. `GetEnumerator` doesn't guarantee correct snapshot and you do need second `ToList` as this is the type of the parameter SaveToFile expects

Comment (fenix2222) [reply src/Stratis.Bitcoin/P2P/PeerAddressManager.cs]: Found a much better alternative. We can use `.Values` collection which has a lock and removes the need to use `Select(v => v.Value)`

Comment (fenix2222) [reply src/Stratis.Bitcoin/P2P/PeerAddressManager.cs]: @dangershony can we merge now?

**Comment by fenix2222** [comment]
@dangershony @bokobza Can one of you merge?

**Pull request merged**